### PR TITLE
DYN-5996: Remove the Arduino project

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

This pull request does:
* add legacy Arduino binaries
* remove Arduino project from solution
* remove Arduino files

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [X] This PR contains no files larger than 50 MB 

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

**Review by aparajit-pratap — APPROVED** [review]

File: src/Libraries/DynamoArduino/Arduino.csproj [hunk 1]
```diff
@@ -20,6 +23,9 @@
     <Reference Include="PresentationFramework" />
     <Reference Include="PresentationCore" />
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Windows.Compatibility" Version="6.0.7" Condition=" '$(TargetFramework)' == 'net6.0-windows' "/>
```
Comment (mjkkirschner) [inline src/Libraries/DynamoArduino/Arduino.csproj:27]: this appears to bring in a million things:
https://www.nuget.org/packages/Microsoft.Windows.Compatibility

1. why do we need it?
2. I vote to remove this project.

Comment (QilongTang) [reply src/Libraries/DynamoArduino/Arduino.csproj:27]: For the last couple years I never see Dynamo actually use this project right?

Comment (sm6srw) [reply src/Libraries/DynamoArduino/Arduino.csproj:27]: Yes, the package is needed for serial port support. Not a problem anymore. I will remove the project instead.

File: extern/legacy_remove_me/bin/DynamoArduino.dll
Comment (pinzart90) [inline extern/legacy_remove_me/bin/DynamoArduino.dll:1]: does this file actually need to contain data ? couldn't it be empty ?

Comment (pinzart90) [reply extern/legacy_remove_me/bin/DynamoArduino.dll:1]: Or do we actually need it to work ?

**Review by pinzart90 — APPROVED** [review]
One question (applies to the resource dll too)

File: extern/legacy_remove_me/bin/DynamoArduino.dll
Comment (sm6srw) [reply extern/legacy_remove_me/bin/DynamoArduino.dll:1]: It needs to have version info etc.

Comment (mjkkirschner) [reply extern/legacy_remove_me/bin/DynamoArduino.dll:1]: and be signed.

Comment (sm6srw) [reply extern/legacy_remove_me/bin/DynamoArduino.dll:1]: Signed, good catch! I forgot about that. I need to do another quick update!

Comment (sm6srw) [reply extern/legacy_remove_me/bin/DynamoArduino.dll:1]: Ok, thanks!

**Pull request merged**